test/powershell/Language/Scripting/I18n.Tests.ps1

```diff
@@ -0,0 +1,180 @@
+
+        import-localizedData mydata -filename foo
+
+        $mydata.string1 | Should Be 'string1 from foo in en-US'
+        $mydata.string2 | Should Be 'string2 from foo in en-US'
+
+        import-localizedData mydata -filename foo -uiculture fr-FR
+
+        $mydata.string1 | Should Be 'string1 from foo in fr-FR'
+        $mydata.string2 | Should Be 'string2 from foo in fr-FR'
+    }
+
+    It 'Import different file base is done correctly' {
+    
+        import-localizedData mydata -basedirectory "${dir}\newbase"
+
+        $mydata.string1 | Should Be 'string1 for en-US under newbase'
+        $mydata.string2 | Should Be 'string2 for en-US under newbase'
+
+        import-localizedData mydata -basedirectory "${dir}\newbase" -uiculture fr-FR
+
+        $mydata.string1 | Should Be 'string1 for fr-FR under newbase'
+        $mydata.string2 | Should Be 'string2 for fr-FR under newbase'
+    }
+
+    It 'Import different file base and file name' {
+    
+        import-localizedData mydata -basedirectory "${dir}\newbase" -filename foo
+
+        $mydata.string1 | Should Be 'string1 for en-US from foo under newbase'
+        $mydata.string2 | Should Be 'string2 for en-US from foo under newbase'
+
+        import-localizedData mydata -basedirectory "${dir}\newbase" -filename foo -uiculture fr-FR
+
+        $mydata.string1 | Should Be 'string1 for fr-FR from foo under newbase'
+        $mydata.string2 | Should Be 'string2 for fr-FR from foo under newbase'
+        }
+
+    It "Import variable that doesn't exist" {    
+
+        import-localizedData mydata2 
+
+        $mydata2.string1 | Should Be 'string1 for en-US'
+        $mydata2.string2 | Should Be 'string2 for en-US'
+    }
+
+    It 'Import bad psd1 file - tests the use of disallowed variables' {
+    
+        $script:exception = $null
+        & { 
+            trap {$script:exception = $_ ; continue }
+            import-localizedData mydata -filename bad
+          } 
+
+        $script:exception.exception.gettype() | Should Be System.management.automation.psinvalidoperationexception
+        }
+
+    It 'Import if psd1 file is done correctly' {
+    
+        import-localizedData mydata -filename if
+
+        if ($psculture -eq 'en-US')
+        {    
+            $mydata.string1 | Should Be 'string1 for en-US in if'
+            $mydata.string2 | Should Be 'string2 for en-US in if'
+        }
+        else
+        {
+            $mydata | Should Be $null
+        }
+    }
+
+    $testData = @(
+        @{cmd = 'data d { @{ x=$(get-command)} }';Expected='*get-command*'},
+        @{cmd = 'data d { if ($(get-command)) {} }';Expected='*get-command*'},
+        @{cmd = 'data d { @(get-command) }';Expected='*get-command*'}
+        )
+
+    It 'Allowed cmdlets checked properly' -TestCase:$testData {
+        param ($cmd, $Expected)
+
+        $script:exception = $null
+        & {
+            trap {$script:exception = $_.Exception ; continue }            
+            iex $cmd
```

Comment:
Fixed in iteration 8

---
In reply to: [77269930](https://github.com/Powershell/Powershell/pull/2092#discussion_r77269930) [](ancestors = 77269930)